[Teo Hao Yu] iP

tobihy opened the pull request:
### Poogie

> Me is Poogie. Poogie love poogieing. Pooogieegieieeeeeeeeee.

Poogie is good at poogieing tasks for you. Poogie is,

- text-based
- easy to learn
- ~~fast~~ **_sanic-fast_** performance

All you need to do is:

1. download it from [here](https://github.com/tobihy/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😃 

And most of all, it's **FREE!**

Fweetures:

- [ ] Managing tasks
- [ ] Managing deadlines
- [ ] Reminders (coming soon...)

Commit:
Add Gradle support

tobihy committed:
Added greet, echo and exit functionality

tobihy committed:
Added add and list functionality

tobihy committed:
Added mark as done and mark as undone functionality

tobihy committed:
Added ToDos, Events and Deadlines

tobihy committed:
Included Automated Text UI Testing

tobihy committed:
Added minimal exception handling

tobihy committed:
Added exception handling for more cases

tobihy committed:
Added support for deleting tasks from the list

tobihy committed:
Edited code to make use of Enums

tobihy committed:
Reformat code according to Java Coding Standard

tobihy committed:
Refactor goodbye method

tobihy committed:
Add save functionality and error handling for more cases

tobihy committed:
Add date and time functionality in some tasks

tobihy committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/Todo.java

tobihy committed:
Add date and time functionality in some tasks

tobihy committed:
Refactor code to extract out closely related code as classes

tobihy committed:
Organise the classes into suitable java packages

tobihy committed:
Add JUnit tests to test the behavior of the code

tobihy committed:
Add JavaDoc comments to the code

tobihy committed:
Tweak the code to comply with a coding standard

tobihy committed:
Give users a way to find a task by searching for a keyword

tobihy committed:
Merge branch 'branch-A-JavaDoc'

tobihy committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/commands/IncorrectCommand.java
#	src/main/java/duke/commands/ListCommand.java

tobihy committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/commands/ListCommand.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java

tobihy committed:
Add JavaDoc comments for FindCommand

tobihy committed:
Add more JavaDoc comments, edit to follow coding standard

tobihy committed:
Add more JavaDoc comments, fix delete, mark and unmark bugs

tobihy committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

tobihy committed:
Automate project builds using Gradle

tobihy committed:
Merge branch 'branch-A-Gradle'

tobihy committed:
Add CheckStyle directory and files

tobihy committed:
Use CheckStyle to detect and rectify coding violations

tobihy committed:
Merge branch 'branch-A-CheckStyle'

lowkaiwei98 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,326 @@
+ */
+public class Parser {
+    // Input formats of date and times
+    private static final DateTimeFormatter dateTimeIn = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private static final DateTimeFormatter dateIn = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private static final DateTimeFormatter timeIn = DateTimeFormatter.ofPattern("HHmm");
```
Constant should be uppercase and underline? I notice there are similar issues in other places.

lowkaiwei98 reviewed (commented):
Overall, i think the naming looks good to me for almost every class and methods, good job 😄

chuashihong commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,58 @@
+
+    private void run() {
+            ui.greetUser();
+            boolean isExit = false;
```
Boolean sounds like boolean, good!

chuashihong commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,29 @@
+     *
+     * @return true if the command is an exit command, false otherwise
+     */
+    public abstract boolean isExit();
```
Perhaps you could let isExit() return False by default and only when it is going to exit, it returns True
You can change this method to become a normal method (instead of abstract class) and returns False by default, then in the **ExitCommand** class, you override it to return True

chuashihong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,326 @@
+ */
+public class Parser {
+    // Input formats of date and times
+    private static final DateTimeFormatter dateTimeIn = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+    private static final DateTimeFormatter dateIn = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private static final DateTimeFormatter timeIn = DateTimeFormatter.ofPattern("HHmm");
+
+    // Output formats of date and times
+    private static final DateTimeFormatter dateTimeOut = DateTimeFormatter.ofPattern("MMM dd yyyy hh:mm a");
+    private static final DateTimeFormatter dateOut = DateTimeFormatter.ofPattern("MMM dd yyyy");
+    private static final DateTimeFormatter timeOut = DateTimeFormatter.ofPattern("hh:mm a");
```
Constant should looks like this:
```
DATE_TIME_IN
TIME_IN
```

chuashihong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,326 @@
+        } else {
+            Parser.Format f = Parser.parseDateTime(at);
+
+            switch (f) {
```
Switch case follows coding standard. Good Job!

chuashihong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,326 @@
+
+        String next = request.substring(5);
+        String[] parsedReq = next.split("/at");
+        String desc = parsedReq[0].strip();
```
A bit ambiguous to this variable name desc, maybe you can refactor its name to description

chuashihong commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,132 @@
+    @Override
+    public String toString() {
+        return "[E]" + super.toString() + " (at: " +
+                (hasDate && hasTime
+                        ? getDateTime()
+                        : hasDate
+                        ? getDate()
+                        : getTime()) + ")";
```
A bit complicated to see what is going to be the outcome
Perhaps you could separate the function before returning it

chuashihong commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,326 @@
+     * @return the command based on the user's input
+     */
+    public Command parseCommand(String input) {
+        if (input.strip().equals("bye")) {
+            return new ExitCommand();
+        } else if (input.strip().equals("list")) {
+            return new ListCommand();
+        } else if (input.startsWith("mark")) {
+            return prepareMark(input);
+        } else if (input.startsWith("unmark")) {
+            return prepareUnmark(input);
+        } else if (input.startsWith("todo")) {
+            return prepareTodo(input);
+        } else if (input.startsWith("deadline")) {
+            return prepareDeadline(input);
+        } else if (input.startsWith("event")) {
+            return prepareEvent(input);
+        } else if (input.startsWith("delete")) {
+            return prepareDelete(input);
+        } else if (input.startsWith("find")) {
+            return prepareFind(input);
+        } else {
+            return new IncorrectCommand("My apologies, but it seems that I do not understand your request.");
+        }
```
Could use Switch Case statement to make it easier to look!

chuashihong commented on src/main/java/duke/task/Todo.java:
```diff
@@ -0,0 +1,55 @@
+     * @return false
+     */
+    @Override
+    public boolean isHasDate() {
+        return false;
+    }
+
+    /**
+     * Checks if a time is present in the todo.
+     *
+     * @return false
+     */
+    @Override
+    public boolean isHasTime() {
+        return false;
+    }
```
the boolean method name is good! But you could make it shorter by renaming it to
```
hasDate
hasTime
```
and still sounds like boolean

chuashihong commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,69 @@
+
+    /** Greets the user */
+    public void greetUser() {
+        String poogie = "    ⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⡿⠛⠉⠉⠉⠉⠉⠉⠉⠛⢿⣿⣿⣿⣿⣿⣿⣿⣿⣿\n" +
+                "    ⣿⣿⣿⣿⣿⣿⣿⣿⣿⠏⠄⢀⣠⣶⣶⣶⣶⣤⡀⠄⠄⠹⣿⣿⣿⣿⣿⣿⣿⣿\n" +
+                "    ⣿⣿⣿⣿⣿⣿⣿⣿⡏⠄⠄⣾⡿⢿⣿⣿⡿⢿⣿⡆⠄⠄⢻⣿⣿⣿⣿⣿⣿⣿\n" +
+                "    ⣿⣿⣿⣿⣿⣿⣿⡿⠃⠄⠄⢿⣇⣸⣿⣿⣇⣸⡿⠃⠄⠄⠸⣿⣿⣿⣿⣿⣿⣿\n" +
+                "    ⣿⣿⣿⣿⣿⡿⠋⠄⠄⠄⠄⠄⠉⠛⠛⠛⠛⠉⠄⠄⠄⠄⠄⠄⠙⣿⣿⣿⣿⣿\n" +
+                "    ⣿⣿⣿⣿⡟⠁⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠈⢿⣿⣿⣿\n" +
+                "    ⣿⣿⣿⡟⠄⠄⠄⠠⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠈⢿⣿⣿\n" +
+                "    ⣿⣿⡟⠄⠄⠄⢠⣆⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⣧⠄⠄⠄⠈⢿⣿\n" +
+                "    ⣿⣿⡇⠄⠄⠄⣾⣿⡀⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⠄⢰⣿⣧⠄⠄⠄⠘⣿\n" +
+                "    ⣿⣿⣇⠄⣰⣶⣿⣿⣿⣦⣀⡀⠄⠄⠄⠄⠄⠄⠄⢀⣠⣴⣿⣿⣿⣶⣆⠄⢀⣿\n" +
+                "    ⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⠏⠄⠄⢸⣿⠇⠄⠄⠹⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿\n" +
+                "    ⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿⣷⣦⣤⣴⣾⣿⣶⣤⣤⣴⣿⣿⣿⣿⣿⣿⣿⣿⣿⣿\n";
```
very cute!

chuashihong reviewed (commented):
Overall the coding standard is not violated too much, except some of the grammar in javaDoc needs to check, like "do something" should change to "Does something..."
Keep up the good work!!!

tobihy committed:
Add a GUI to Duke. Use the JavaFX technology to implement the GUI.

mazx4960 commented on src/main/java/duke/commands/ExitCommand.java:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
method name is the same as the variable in duke. could perhaps be named as canExit

mazx4960 commented on src/main/java/duke/commands/FindCommand.java:
```diff
@@ -0,0 +1,46 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
method name is the same as the variable in duke. could perhaps be named as canExit

mazx4960 commented on src/main/java/duke/commands/IncorrectCommand.java:
```diff
@@ -0,0 +1,39 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
method name is the same as the variable in duke. could perhaps be named as canExit

mazx4960 commented on src/main/java/duke/commands/ListCommand.java:
```diff
@@ -0,0 +1,30 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
method name is the same as the variable in duke. could perhaps be named as canExit

mazx4960 commented on src/main/java/duke/commands/MarkCommand.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
method name is the same as the variable in duke. could perhaps be named as canExit

mazx4960 commented on src/main/java/duke/commands/UnmarkCommand.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
method name is the same as the variable in duke. could perhaps be named as canExit

mazx4960 commented on src/main/java/duke/commands/IncorrectCommand.java:
```diff
@@ -0,0 +1,39 @@
+/**
+ * Represents an incorrect command.
+ */
+public class IncorrectCommand extends Command {
```
Instead of using a class for an incorrect command, perhaps you could throw an exception instead

mazx4960 commented on src/test/java/duke/parser/ParserTest.java:
```diff
@@ -0,0 +1,59 @@
+    }
+
+    @Test
+    public void parse_emptyInput_returnsIncorrectCommand() {
```
informative and great test method name!

mazx4960 reviewed (commented):
Overall, the code follows the java coding standard closely. Method and variable names were also well defined, with the exception of some.

tobihy committed:
Change duke and user profile pictures and edit FXML files

tobihy committed:
Add missing JavaDoc comments and change names of constants

tobihy committed:
Merge branch 'branch-Level-10'

tobihy committed:
Add CSS styling to more elements

tobihy committed:
Use Assertions

Code uses many if-else statements to control the flow of operations.

Assertions help to document important assumptions that should hold at
various points in the code.

Using Assertions is preferable over Exceptions as Assertions are used to
catch errors before runtime.

tobihy committed:
Improve code quality

Classes contain some deeply nested if-else statements and excessively
long methods

This can negatively impact code readability

Let's,
* Refactor the long methods
* Restructure code

tobihy committed:
Use Streams in find command

tobihy committed:
Use Assertions

Code uses many if-else statements to control the flow of operations.

Assertions help to document important assumptions that should hold at
various points in the code.

Using Assertions is preferable over Exceptions as Assertions are used to
catch errors before runtime.

tobihy committed:
Merge branch 'master' of github.com:tobihy/ip

tobihy committed:
Merge master into branch-A-CodeQuality

tobihy committed:
Merge branch 'master' into branch-A-Streams

tobihy committed:
Merge pull request #2 from tobihy/branch-A-CodeQuality

Improve code quality

tobihy committed:
Merge branch 'master' of github.com:tobihy/ip

tobihy committed:
Merge branch 'master' into branch-A-Streams

tobihy committed:
Merge pull request #3 from tobihy/branch-A-Streams

Find command: use Streams in find method

Find method uses a for-loop to traverse the TaskList and adds tasks with matching keywords.

Use streams to
* Simplify the implementation of the filter method

This improves the code readability.

tobihy committed:
Merge branch 'master' of github.com:tobihy/ip

tobihy committed:
Add ability to easily edit tasks

tobihy committed:
Merge branch 'branch-C-Update'

tobihy committed:
Improve code quality: extract various exceptions

Exceptions in the Parser class have similar structure and behaviour.

Using a command class IncorrectCommand is inaccurate, as errors
are not executed and hence should not be handled using a command class.

Let's pull out errors in the parser that have behaviour in common into
new exceptions.

Using inheritance for the subclasses of DukeException over composition
is preferred as the exceptions have different error messages.

tobihy committed:
Improve the GUI to make it more polished

tobihy committed:
Merge branch 'branch-A-BetterGui'

tobihy committed:
Add spacing between dialog and display photo and refactor code

tobihy committed:
Fix some bugs

tobihy committed:
Add a user guide and fix directory issues in Storage.java

tobihy committed:
Reformat User Guide for better readability

tobihy committed:
Change archiveBaseName in build.gradle

tobihy committed:
Change profile photos used in the UI

tobihy committed:
Add JUnit tests

tobihy committed:
Clean up unused files and Java checkstyle warnings

tobihy committed:
Change directory path of save data folder

tobihy committed:
Fix file path issues

tobihy committed:
Set theme jekyll-theme-minimal

tobihy committed:
Edit README.md

tobihy committed:
Merge branch 'master' of github.com:tobihy/ip

tobihy committed:
Edit README.md

tobihy committed:
Fix bug for EditCommand
